decls/haskell_rules.bzl

```diff
@@ -188,6 +188,7 @@ haskell_library = prelude_rule(
             "linker_flags": attrs.list(attrs.arg(), default = []),
             "platform": attrs.option(attrs.string(), default = None),
             "platform_linker_flags": attrs.list(attrs.tuple(attrs.regex(), attrs.list(attrs.arg())), default = []),
+            "use_same_package_name": attrs.bool(default = False),
```

Comment:
Maybe this could be an optional `package_id` attribute instead? The default behavior would still be the current behavior, i.e. equivalent with `use_same_package_name = False`, but then when `package_id` is set it's a bit more flexible and allows users direct control over the package id if needed. (For reference [Bazel rules_haskell](https://api.haskell.build/haskell/defs#haskell_library-package_name) has something like this too.)

Also, to clarify. IIUC this actually directly influences `-this-unit-id`, correct? So, maybe the name `package_id` is more appropriate than `package_name`, or maybe it should directly be called `unit_id`?